src/plzinit/go.go

```diff
@@ -0,0 +1,67 @@
+package plzinit
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/thought-machine/please/src/cli"
+)
+
+// golangConfig prompts the user and returns the go config to add to the .plzconfig
+func golangConfig(dir string) string {
+	//TODO(jpoole): make this check that go is in the please path
+	//TODO(jpoole): detect if we're in a go project and suggest they enable GO_BUILD_COMPAT to help migration
+
+	goModule, ok := findGoModule(dir)
+
+	if !ok && !cli.PromptYN("Would you like to setup golang in this repository", false) {
+		return ""
+	}
+
+	config := "\n[go]\n"
+
+	goRoot, err := cli.Prompt("Do you want to set GOROOT? Otherwise please will use `go env GOROOT`", os.Getenv("GOROOT"))
```

Comment:
I'm not sure this is right. `GOROOT` != `go env GOROOT`. As we discussed earlier, `GOROOT` does not have to be set, the go tool can just figure it out.

Also, a custom `GOROOT` is probably a local config, chances are that in CI or on other machines it will be different.

Personally, I would not prompt for setting `GOROOT`. If it comes from the environment, Please already handles that well (fixed in #1195). Otherwise it should come from the go tool, thus should not be set at all.

Reply:
Maybe set this in `.plzconfig.local`? But that feels unnecessary too.

Reply:
> As we discussed earlier, GOROOT does not have to be set, the go tool can just figure it out.

It's how we find the golang SDK. If GOROOT it not set, we look in `$(go env GOROOT)/go` for the `.a`s

Very valid points though. I think I should just check that `$(which go)` is somewhere in the `path` config option and ask if they would like to add it.

Reply:
Forgot about the stdlib. That's valid.

But even then, the current goroot "detection" from the environment should be fine. Hard coding it in the config makes the project less portable.